extensions/sql-migration/src/dialog/restartMigration/restartMigrationDialog.ts

```diff
@@ -0,0 +1,177 @@
+
+		const getStorageAccountResourceGroup = (storageAccountResourceId: string): azureResource.AzureResourceResourceGroup => {
+			return {
+				id: getFullResourceGroupFromId(storageAccountResourceId!),
+				name: getResourceGroupFromId(storageAccountResourceId!),
+				subscription: this._serviceContext.subscription!
+			};
+		};
+		const getStorageAccount = (storageAccountResourceId: string): azureResource.AzureGraphResource => {
+			const storageAccountName = getResourceName(storageAccountResourceId);
+			return {
+				type: 'microsoft.storage/storageaccounts',
+				id: storageAccountResourceId!,
+				tenantId: savedInfo.azureTenant?.id!,
+				subscriptionId: this._serviceContext.subscription?.id!,
+				name: storageAccountName,
+				location: savedInfo.location!.name,
+			};
+		};
+
+		const sourceLocation = migration.properties.backupConfiguration?.sourceLocation;
+		if (sourceLocation?.fileShare) {
+			savedInfo.networkContainerType = NetworkContainerType.NETWORK_SHARE;
+			const storageAccountResourceId = migration.properties.backupConfiguration?.targetLocation?.storageAccountResourceId!;
+			savedInfo.networkShares = [
+				{
+					password: '',
+					networkShareLocation: sourceLocation?.fileShare?.path!,
+					windowsUser: sourceLocation?.fileShare?.username!,
+					storageAccount: getStorageAccount(storageAccountResourceId!),
+					resourceGroup: getStorageAccountResourceGroup(storageAccountResourceId!),
+					storageKey: ''
+				}
+			];
+		} else if (sourceLocation?.azureBlob) {
+			savedInfo.networkContainerType = NetworkContainerType.BLOB_CONTAINER;
+			const storageAccountResourceId = sourceLocation?.azureBlob?.storageAccountResourceId!;
+			savedInfo.blobs = [
+				{
+					blobContainer: {
+						id: getBlobContainerId(getFullResourceGroupFromId(storageAccountResourceId!), getResourceName(storageAccountResourceId!), sourceLocation?.azureBlob.blobContainerName),
+						name: sourceLocation?.azureBlob.blobContainerName,
+						subscription: this._serviceContext.subscription!
+					},
+					lastBackupFile: getMigrationModeEnum(migration) === MigrationMode.OFFLINE ? migration.properties.offlineConfiguration?.lastBackupName! : undefined,
+					storageAccount: getStorageAccount(storageAccountResourceId!),
+					resourceGroup: getStorageAccountResourceGroup(storageAccountResourceId!),
+					storageKey: ''
+				}
+			];
+		}
+
+		stateModel.restartMigration = true;
+		stateModel.savedInfo = savedInfo;
+		stateModel.serverName = serverName;
+		return stateModel;
+	}
+
+	public async openDialog(dialogName?: string) {
+		const locations = await getLocations(
+			this._serviceContext.azureAccount!,
+			this._serviceContext.subscription!);
+
+		const targetInstance = await getMigrationTargetInstance(
+			this._serviceContext.azureAccount!,
+			this._serviceContext.subscription!,
+			this._migration);
+
+		let location: azureResource.AzureLocation;
+		locations.forEach(azureLocation => {
+			if (azureLocation.name === targetInstance.location) {
```

Comment:
don't remember if these are always lowercase or not... maybe case insensitive comparison just in case?

Reply:
This is not part of the PR, just a file rename